core/templates/pages/contributor-dashboard-admin-page/contributor-admin-dashboard-page.component.ts

```diff
@@ -189,6 +190,13 @@ export class ContributorAdminDashboardPageComponent implements OnInit {
       );
   }
 
+  putEnglishLanguageNameAtFront(language: string): string {
+    // Check if the language contains the expected format 'abc (something)'.
```

Comment:
Looks nice, much easier to understand!

One question -- what happens if the string fails the regex? Should have a test for that since it could happen due to e.g. programmer error / incorrect assumptions / etc. or something, and we don't want it to take down the site.

Also are we assuming that the language name has no spaces? That might not be true for the language or its translated version. A simpler regex might be just: "something (something)" where either something can contain spaces -- I think it's a safe bet that language strings won't have parens :P

Reply:
> what happens if the string fails the regex? Should have a test for that since it could happen due to e.g. programmer error / incorrect assumptions / etc. or something, and we don't want it to take down the site.

Have added the test case.

> Also are we assuming that the language name has no spaces? That might not be true for the language or its translated version...

Changed the regex.